### DISCUSSION TITLE
video duration

### DISCUSSION THREAD

**Discussion by kdy1968** [Geektoolkit/Dynaframe3#174] [Ideas]
tried out your project very nice, appreciate your hard work for those that have worked on it. 
things i would like to see, more transistions modes...not sure if the pi would support it..

i was planning on using it as a digital piture frame ..but was more interested in it is ability to play back video...the way its currently setup duration effects both pictures and video playback, thus video get cut off, be nice if the video would play in its entirety before loading next item...duration only  affecting picture display, when video is playing have transition when play back completes...or just before..
ill be watching where you take this project...

**Comment by Geektoolkit** [comment]
I'm working on a new version called Dynaframe Pro which will have the feature of being able to play the full videos. I'm going to leave this project to the community to work on and continue with, but the Dynaframe pro project will be more of a supported product backed by a simple and inexpensive patreon (only one time purchase needed, but allows for continued support as I update it).  That product will have features such as what you listed, web uploads, and also be provided as an image so that it'll basically be 'download and run'.  You wont even need a keyboard or mouse to hook up to the frame to use it. I'm working hard on getting that release out in January

**Comment by Archonw** [comment]
Can't wait for it!

**Comment by broswell** [comment]
Also looking forward to videos running to the end.
Did we lose the integration with MQTT in the upgrade from the original to this?  Will it be in the next version?

**Reply by Geektoolkit** [reply]
We did lose it...I never liked how it was done in the original and want to include a much more robust and useful MQTT setup (querying for currently playing info, changing settings, home assistant auto discovery). That's planned for PRO but won't be in the initial release (expected by end of January). 
   If you have things you want in MQTT please let me know what features you used and I can make sure they make it into the implementation or at least give it my best shot

**Reply by broswell** [reply]
Intended use is for a museum display.  (https://museum.syssrc.com)

1) I have wireless (433Mhz) motion sensors.  When someone walks into the area, turn on the display.  Right now I do that with rtl_433 to MQTT -> HomeAssistant to a wifi plug.  I think there is a better way via HDMI.

2) If no docent, slides and videos play.  Ideally be able to set the time for each slide like PowerPoint

3) When I am giving a tour, I use the remote controls (on regular digital picture frames) to advance to the proper slide.  Remote control is also 433Mhz -> rtl_433 -> MQTT.  Ideally forward and back like the cheap picture frames.    I am not using the cheap picture frame due to the size needed for the display.  We will be using a 55" TV hooked to the Pi.

Thanks

**Reply by Geektoolkit** [reply]
OMG...your museum is awesome looking! Also to realize my softwares in a museum is really amazing to me! 
  That's a really amazing setup. I'll be able to figure out something to help get that scenario working and want to make sure this works great for you. Helping a non profit museum that helps others appreciate and learn about tech is important to me.
  Until I can get through my current work items is it possible for you to leverage the http api? I have it documented in the wiki I believe. It allows for forward, back and go to a certain image. You can also select playlists. Basically the web UI uses http to do everything.
   Dynaframe pros http api is much more consistent. 
   This is awesome!

**Reply by broswell** [reply]
Thanks.  Setting up a Cray Room (Cray-1, Cray-2) which we just got from Digibarn.com.  We have other Crays.  For now, I have to use another solution since I can't figure out the video duration.  The videos are of varying lengths (1-7 minutes) while the need about 20 seconds.

I just put together simple HTML pages with the slides and videos and I am using a meta-refresh to auto advance.  have not conquered the remote :(
